src/main/java/by/exadel/internship/repository/InternshipRepository.java

```diff
@@ -2,16 +2,32 @@
 
 import by.exadel.internship.entity.Internship;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Modifying;
 import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 
 import java.util.List;
+import java.util.Optional;
 import java.util.UUID;
 
 @Repository
 public interface InternshipRepository extends JpaRepository<Internship, UUID> {
 
-    @Query("SELECT distinct i FROM Internship i LEFT JOIN FETCH i.skills")
+    List<Internship> findAllByDeletedTrue();
+    List<Internship> findAllByDeletedFalse();
+    Optional<Internship> findAllByIdAndDeletedFalse(UUID internshipId);
+    Optional<Internship> findByIdAndDeletedTrue(UUID internshipId);
+
+    @Modifying
+    @Query("update Internship i set i.deleted = false WHERE i.id = :internshipId")
+    void activateInternshipById(@Param("internshipId") UUID internshipId);
+
+    @Query("SELECT distinct i FROM Internship i LEFT JOIN FETCH i.skills WHERE i.deleted = false")
     List<Internship> findAllWithSkill();
 
+    @Modifying
+    @Query(value = "UPDATE internship SET inship_deleted=true WHERE inship_id= :internshipId",
```

Comment:
Why do you leave this query as native? Try to use single strategy.

Reply:
We left this method to delete only one entity.